### PR TITLE
feat: Allow passing provider parameters

### LINKED ISSUE #10: [Feature Requests] Allow passing provider parameters
Given the provider like this:

```json
{
  "name": "twitter",
  "services": [
    {
      "id": "default",
      "baseUrl": "https://api.twitter.com"
    }
  ],
  "parameters": [
    {
      "name": "accessToken",
      "description": "User-authorized OAuth 2.0 access token"
    }
  ],
  "defaultService": "default"
}
```

I would like to perform a query with `accessToken` passed in input instead of being hard-coded with deployment. It would probably make sense to allow passing `parameters` in `options` argument. I don't think it needs to be typed, though not sure if GraphQL allows passing just untyped record.

### PR DESCRIPTION
**Pull request by jnv**
Fixes #10.

### CODE REVIEW

File: src/schema.types.spec.ts [hunk 1]
```diff
@@ -178,8 +197,37 @@ describe('schema.types', () => {
...
+  });
+
+  describe('generateUseCaseProviderParametersFields', () => {
+    it('generates provider parameters for configured providers', () => {
+      expect(
+        generateUseCaseProviderParametersFields(
+          ['superface', 'mock'],
+          providers,
+        ),
+      ).toMatchInlineSnapshot(`
```
Comment (freaz) [inline src/schema.types.spec.ts:214]: Please use standalone snapshot as other test cases.

File: src/schema.types.ts [hunk 2]
```diff
@@ -236,10 +249,50 @@ export function generateUseCaseOptionsInputType(
...
+  }
+
+  debug(
+    'generateUseCaseProviderParametersFields found provider parameters %o',
+    parameterNames,
+  );
+
+  const fields: GraphQLInputFieldConfigMap = {};
+  for (const parameterName of parameterNames.values()) {
+    fields[parameterName] = { type: GraphQLString };
```
Comment (freaz) [inline src/schema.types.ts:290]: here should be parameter [description[(https://github.com/superfaceai/ast-js/blob/dev/src/interfaces/providerjson/providerjson.schema.json#L35-L37) set.

And as I mention in #10 also add supported providers to description as well.

That will improve user experience a lot.

image

**Review by freaz — COMMENTED** [review]
Looks promising, just not happy about parameters missing any descriptions and will be hard to use.

**Comment by freaz** [comment]
Oh and please fiy commit message formats

**Comment by jnv** [comment]
@freaz Thanks for the review!

- Regarding parameter descriptions, I wanted to implement it as a follow-up, I didn't see it as a blocker; I can unlink the related issue and keep it open until the descriptions are handled there (I plan to introduce a multimap strcuture for that so it properly handles overlapping parameters from multiple providers)
- Regarding commit messages, I planned to squash the commits, didn't plan to expose them in the history.

**Comment by freaz** [comment]
I see, then it is fine, I just didn't know.

**Review by freaz — APPROVED** [review]

**Pull request merged**